0035-ilp-over-http/0035-ilp-over-http.md

```diff
@@ -52,7 +59,72 @@ Content-Type: application/octet-stream
+
+The sender SHOULD ensure there are multiple attempts to deliver the packet to the peer before the ILP Prepare expires.
+
+### Async ILP Fulfill/Reject Reply
+
+#### Request
+
+```http
+POST /ilp HTTP/x.x
+Host: example.com
+Content-Type: application/octet-stream
+Authorization: Bearer zxcljvoizuu09wqqpowipoalksdflksjdgxclvkjl0s909asdf
+Request-Id: 42ee09c8-a6de-4ae3-8a47-4732b0cbb07b
+Idempotency-Key: 6ff99499-008e-4499-8644-048450627496
+
+< Body: Binary OER-Encoded ILP Fulfill or Reject Packet >
+```
+
+- **Path** &mdash; A connector MAY specify any HTTP path for their peer to send ILP packets to.
+- **Host Header** &mdash; The standard [HTTP Host Header](https://tools.ietf.org/html/rfc2616#section-14.23) indicating the domain of the HTTP server the Request is sent to.
+- **Content-Type Header** &mdash; MUST be set to `application/octet-stream`.
+- **Request Id Header** &mdash; Request ID from the corresponding ILP Prepare, which is a UUIDv4.
+- **Idempotency Key Header** &mdash; _Optional_. UUIDv4 to uniquely identify this packet.
+- **Body** &mdash; ILP Packet encoded using OER, as specified in [RFC 27: Interledger Protocol V4](./0027-interledger-protocol-4/0027-interledger-protocol-4.md).
+
+#### Response
+
+```http
+HTTP/x.x 200 OK
+```
+
+If the request ID doesn't correspond to an in-flight ILP Prepare, the connector should ignore it and return an error:
+
+```http
+HTTP/x.x 400 Bad Request
+```
+
+#### Retry Behavior
+
+If the recipient of the ILP Fulfill/Reject responds with an `5xx` or `409 Conflict` status, or no HTTP response is received within a given timeout, the sender of the ILP Fulfill/Reject SHOULD retry sending the packet with the same idempotency key.
+
+The sender of the ILP Fulfill/Reject MUST conclude retrying after receiving a response with a `2xx` status or `4xx` error.
+
+The sender SHOULD ensure there are multiple attempts to deliver the reply packet to the peer before the corresponding ILP Prepare expires.
+
+### Idempotence
+
+Every Interledger packet corresponds to a transaction that may affect financial accounting balances. If a request fails, such as due to a network connection error, retrying ILP requests and responses with [idempotence](https://en.wikipedia.org/wiki/Idempotence) can prevent balance inconsistencies between peers.
+
+When a connector begins processing an incoming packet with an idempotency key it has not already tracked, it MUST persist that key. If a subsequent request of the same type (ILP Prepare, or ILP Fulfill/Reject) is encountered with the same idempotency key, the packet should be ignored, and the connector should respond with the successful status. For safety, the connector MUST persist each idempotency key until some amount of time after the corresponding ILP Prepare expires so it doesn't accidentally process a duplicate ILP Prepare packet.
```

Comment:
if we want to make this nonbreaking we should clarify that idempotence is a feature of asynchronous mode only. If we want to make it nonbreaking and also add idempotence to synchronous mode then we should have a response code/header of some kind that makes it clear the idempotency key will be honored

Reply:
Since ILP Prepare retries must be negotiated out of band, isn't this still non-breaking? If the peers don't retry ILP Prepares, the recipient of the ILP Prepare doesn't need to handle idempotent requests. Can I clarify that better?

Reply:
Added an explanation here that hopefully better clarifies this: https://github.com/interledger/rfcs/compare/09916ea8c1232f846610940066298f052f140c8f...8b43dafce2c84d073bc70afacf396f700d4ff8f5#diff-ccfda9b7df50a9cfd8618bb9c08e360aR126-R132